build/xtask/src/dist.rs

```diff
@@ -1224,6 +1215,31 @@ fn allocate_all(
     Ok(allocs)
 }
 
+fn allocate_k(
+    region: &str,
+    size: u32,
+    avail: &mut Range<u32>,
+) -> Result<Range<u32>> {
+    // Our base address will be larger than avail.start if it doesn't meet our
+    // minimum requirements. Round up.
+    let base = (avail.start + 15) & !15;
+
+    if base >= avail.end || size > avail.end - base {
```

Comment:
Nit – could you do
```
    let end = base + size;
    if end > avail.end {
        ...
```
instead of checking both conditions?

This would produce different results if `base == avail.end && size == 0`, but I'm not sure what answer is correct there...

Reply:
...I _believe_ that yours is equivalent, yes.

I also note that `Range::contains` has been stabilized in 1.35.0, so we could probably make this even less silly.